src/app/api/user/registerUser/route.js

```diff
@@ -0,0 +1,28 @@
+import { NextResponse } from 'next/server'
+import dbConnect from '../../../../../utils/dbConnect';
+import User from '../../../../../models/User';
+
+export const POST = async(req) => {
+  try {
+    await dbConnect();
+
+    const body = await req.json();
+
+    const userExists = await User.find({email: body.email});
+
+    if (userExists.length > 0) return NextResponse.json({error: "User with this email already exists"}, {status: 409});

```

Comment:
Similar to my other feedback, have you thought about a simple error message like "Oops! User already exists!"?

This way, it's not known what information is used to determine if a user exists.